public/app/features/datasources/components/DataSourceAddButton.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import { LinkButton } from '@grafana/ui';
+import { contextSrv } from 'app/core/core';
+import { AccessControlAction } from 'app/types';
+
+import { useDataSourcesRoutes } from '../state';
+
+export function DataSourceAddButton() {
+  const canCreateDataSource = contextSrv.hasPermission(AccessControlAction.DataSourcesCreate);
+  const dataSourcesRoutes = useDataSourcesRoutes();
+
+  return (
+    <LinkButton icon="plus" href={dataSourcesRoutes.New} disabled={!canCreateDataSource}>
```

Comment:
nit: I'm not a designer but rather than disabling a button it might be nicer to hide it entirely so it doesn't cause confusion for a user that doesn't have permission to do something.

Reply:
Done